beta/src/pages/learn/adjusting-effect-dependencies.md

```diff
@@ -0,0 +1,1345 @@
+
+> **Code examples in this section don't work yet!** They rely on a [proposed `useEvent` API](https://github.com/reactjs/rfcs/pull/220).
+
+In this example, the `ChatRoom` component sends an analytics log when the component is added to the screen:
+
+```js {3-5}
+function ChatRoom() {
+  useEffect(() => {
+    post('/analytics/event', {
+      eventName: 'visit_chat',
+    });
+  }, []); // ✅ All dependencies declared
+  // ...
+}
+```
+
+Let's say you add a `roomId` prop and want to include it in the analytics log. You'll have to include it in the Effect dependencies. This ensures that you'll send an extra analytics log every time the user opens a *different* room:
+
+```js {1,5,7}
+function ChatRoom({ roomId }) {
+  useEffect(() => {
+    post('/analytics/event', {
+      eventName: 'visit_chat',
+      roomId: roomId
+    });
+  }, [roomId]); // ✅ All dependencies declared
+  // ...
+}
+```
+
+Now let's say that you want to include some additional information about the application's state into your analytics. For example, you may want to include the user's current notification count into the analytics event:
+
+```js {1,6}
+function ChatRoom({ roomId, notificationCount }) {
+  useEffect(() => {
+    post('/analytics/event', {
+      eventName: 'visit_chat',
+      roomId: roomId,
+      notificationCount: notificationCount
+    });
+  }, [roomId]); // 🔴 React Hook useEffect has a missing dependency: 'notificationCount'
+  // ...
+}
+```
+
+You're using `notificationCount` inside your Effect, so you must add it to the dependency list. However, if you do that, your Effect will re-run whenever the `notificationCount` changes. This doesn't make sense: a change to the number of notification doesn't mean that the user visited the chat again.
+
+**From the user's perspective, "visiting the chat" is similar to an event: it happens at a particular moment in time, and the user can distinguish when it happened. To express this in code, declare it as an *Event function:***
+
+```js {2,8,11}
+function ChatRoom({ roomId, notificationCount }) {
+  const handleVisit = useEvent((visitedRoomId) => {
+    post('/analytics/event', {
+      eventName: 'visit_chat',
+      roomId: visitedRoomId,
+      notificationCount: notificationCount
+    });
+  });
+
+  useEffect(() => {
+    handleVisit(roomId);
+  }, [roomId]); // ✅ All dependencies declared
+  // ...
+}
+```
+
+With this code, the first analytics log will include the initial `roomId` and the initial `notificationCount`. Then, if the `roomId` changes, your Effect will call `handleVisit` again and send a new analytics log with that `roomId` and the current `notificationCount`. However, if the `notificationCount` count changes alone, nothing will happen. **Code inside Event functions is not reactive, so editing `handleVisit` won't make your Effect run more often.**
```

Comment:
although you can kind of infer it, I think this needs some explicit commentary on why roomId is an argument instead of captured